Adding enforce restrictions option

romeupalos opened the pull request:
### Description
This change creates an option to enforce the restriction options in the XML schema in the strong-soap client.
It's useful in test environments to catch possible bugs that would only show up in production when the server side is validating the input request.

After the commit, it's possible to set the client to enforce the following XML restrictions:
- enumeration
- fractionDigits
- length
- maxExclusive
- maxInclusive
- maxLength
- minExclusive
- minInclusive
- minLength
- pattern
- totalDigits
- whiteSpace


#### Related issues
- connect to #51 (partially fixes it)
<!--
Please use the following link syntaxes:


- connect to #49 (to reference issues in the current repository)
- connect to strongloop/loopback#49 (to reference issues in another repository)
-->

### Checklist

<!--
- Please mark your choice with an "x" (i.e. [x], see
https://github.com/blog/1375-task-lists-in-gfm-issues-pulls-comments)
- PR's without test coverage will be closed.
-->

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

slnode commented:
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

dhmlau commented:
@romeupalos , thanks for your PR.  Could you please fix the commit message?  Here is the commit linter error:
```
**  Linting commit logs
**
**  1 problems found:
**    d92701d - Adding enforce restrictions option: line 3 longer than 72 characters.
```

Please refer to our commit message guideline: https://loopback.io/doc/en/contrib/git-commit-messages.html.  Thanks! 

Just wondering... did you extract the message to messages.json by running `slt-globalize -e` against this repo?  I'd like to make sure the string is properly extracted.

romeupalos commented:
Hello @dhmlau,
I fixed the commit message.. sorry I didn't realize it was not compliant with the guideline.
Also I didn't use strong-globalize at first... great project.
Now I've extracted the message with it... It seems a intl/zz folder was created... is it intended to be committed?

Please, let me know if I need to change anything else.
